tests/codex/helpers/mockreservations.nim

```diff
@@ -0,0 +1,34 @@
+import pkg/chronos
+import pkg/codex/sales
+import pkg/codex/stores
+import pkg/questionable/results
+
+type
+  MockReservations* = ref object of Reservations
+    createReservationThrowBytesOutOfBoundsError: bool
+
+func new*(
+    T: type MockReservations,
+    repo: RepoStore
+): MockReservations =
+  ## Create a mock clock instance
+  {.cast(noSideEffect).}:
```

Comment:
I am not sure why the compiler was complaining about `newAsyncLock` having side-effects here but not in the `Reservations.new()` 😳 I have solved this with this workaround, but not sure if this is really the solution... Can somebody let me know please?

Reply:
Chance `func` to `proc`.

`func` is an alias for `proc {.noSideEffect.}`

Reply:
Ohhhh fu** me 🙈 I missed that it is `func` 🙈 Thanks!

Reply:
⌨️🧠🪑

Reply:
Haha yeah, today is that sort of day... 🙄